File: medplum-java/medplum-model/src/main/java/com/fhir/Identifier.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableIdentifier.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Identifier {
  Optional<CodeableConcept> type();

  Optional<String> id();

  Optional<Period> period();

  Optional<Reference> assigner();

  Optional<uri> system();

  Optional<IdentifierUse> use();

  Optional<List<Extension>> extension();

  Optional<String> value();

  static ImmutableIdentifier.Builder builder() {
    return ImmutableIdentifier.builder();
  }
}

File: medplum-typescript/src/model/fhir/Identifier.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { IdentifierUse } from "./IdentifierUse";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { uri } from "./uri";

export interface Identifier {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    use: IdentifierUse | null | undefined;
    type: CodeableConcept | null | undefined;
    system: uri | null | undefined;
    value: string | null | undefined;
    period: Period | null | undefined;
    assigner: Reference | null | undefined;
}
